src/rules/no-deprecated-element.ts

```diff
@@ -0,0 +1,124 @@
+import { ESLintUtils, TSESTree } from "@typescript-eslint/utils";
+import { RuleContext } from "@typescript-eslint/utils/dist/ts-eslint";
+
+type MessageIds = "noDeprecatedElement" | "noDeprecatedElement_replacement";
+type Deprecated = {
+  element: string;
+  replaceWith?: string;
+};
+export type NoDeprecatedElementOptions = [
+  {
+    deprecate?: Deprecated[];
+  }
+];
+
+const buildDeprecationMap = (
+  context: Readonly<RuleContext<MessageIds, NoDeprecatedElementOptions>>
+): {
+  [key: string]: Deprecated;
+} => {
+  const replacements = context.options[0]?.deprecate;
+  const replacementMap =
+    replacements?.reduce((acc, curr) => {
+      return {
+        ...acc,
+        [curr.element]: {
+          ...curr,
+        },
+      };
+    }, {}) ?? {};
+  return replacementMap;
+};
+
+const messages = {
+  noDeprecatedElement: "<{{element}}> is deprecated",
+  noDeprecatedElement_replacement:
+    "<{{element}}> is deprecated, use <{{replaceWith}}> instead.",
+};
+
+export default ESLintUtils.RuleCreator(
+  (name) =>
+    `https://github.com/buildertrend/eslint-plugin-enterprise-extras/blob/main/docs/${name}.md`
+)<NoDeprecatedElementOptions, MessageIds>({
+  name: "no-deprecated-element",
+  meta: {
+    type: "suggestion",
+    fixable: "code",
+    docs: {
+      recommended: "error",
+      description: "Deprecated elements should not be used",
+    },
+    messages,
+    schema: [
+      {
+        type: "object",
+        properties: {
+          deprecate: {
+            type: "array",
+            items: {
+              allOf: [
+                {
+                  type: "object",
+                  properties: {
+                    element: { type: "string" },
+                    replaceWith: { type: "string" },
+                  },
+                  required: ["element"],
+                  additionalProperties: false,
+                },
+              ],
+            },
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [{}],
+  create: function (context) {
+    const deprecationMap = buildDeprecationMap(context);
+    const keysToDeprecate = Object.keys(deprecationMap);
+    return {
+      JSXElement: (jsxElement: TSESTree.JSXElement) => {
+        const startingIdentifier = jsxElement.openingElement
+          .name as TSESTree.JSXIdentifier;
+
+        if (!keysToDeprecate.includes(startingIdentifier.name)) {
```

Comment:
Don't know what impact this would be on performance, but doing `startingIdentifier.name in deprecationMap` would be constant time instead of `.includes`. Only may be important if you have a lot of deprecations; this runs on every JSX symbol.

Reply:
Will update